samples/snippets/src/main/java/pubsub/PublishAvroRecordsExample.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pubsub;
+
+// [START pubsub_publish_avro_records]
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.pubsub.v1.TopicAdminClient;
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.Encoding;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.TopicName;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import utilities.State;
+
+public class PublishAvroRecordsExample {
+
+  public static void main(String... args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    // Use a topic created with an Avro schema.
+    String topicId = "your-topic-id";
+
+    publishAvroRecordsExample(projectId, topicId);
+  }
+
+  public static void publishAvroRecordsExample(String projectId, String topicId)
+      throws IOException, ExecutionException, InterruptedException {
+
+    Encoding encoding = null;
+
+    TopicName topicName = TopicName.of(projectId, topicId);
+
+    // Get the topic encoding type.
+    try (TopicAdminClient topicAdminClient = TopicAdminClient.create()) {
+      encoding = topicAdminClient.getTopic(topicName).getSchemaSettings().getEncoding();
+    }
+
+    // Create an object of an avro-tools-generated class.
+    State state = State.newBuilder().setName("Alaska").setPostAbbr("AK").build();
+
+    Publisher publisher = null;
+
+    block:
```

Comment:
what's this?

Reply:
Naming the try catch block so we can exit from this block entirely from the Switch case block.